### DISCUSSION TITLE
Issue from Template: Labels not applying for non-write users

### DISCUSSION THREAD

**Discussion by sarina** [community/community#22510] [Projects and Issues]
<p>Hi,</p>
<p>I have a <a href="https://raw.githubusercontent.com/openedx/.github/master/.github/ISSUE_TEMPLATE/depr-ticket.yml" rel="noopener nofollow ugc">Issue Template</a> defined in my org’s <code>.github</code> repo and cascading down to children repos. The template states a default label that should be applied to the issue upon creation; children repos do have this label defined. This label creation works for members who have write access to the repo, but not for those who don’t. Is this a known limitation (I can’t find any documentation on the subject)? Am I doing something wrong?</p>
<ul>
<li>Issue from a non-write-access user didn’t get DEPR label applied: openedx/public-engineering/issues/17</li>
<li>Issue from a write-access user did get DEPR label applied: openedx/public-engineering/issues/18</li>
</ul>
<p>This label is meant to kick off some GH actions. I can change to parsing the title of the Issue, but this seems kinda silly… the whole point of the issue template providing default label is so the issue is automatically routed correctly when you use the template.</p>
<p>Thanks for your time!</p>

**Comment by jfullstackdev** [comment]
<p>maybe it’s related to label limit for non-write access,</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/47092464/181071560-9d83f4be-4b4a-42d8-b348-f600a19b95bc.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels" target="_blank" rel="noopener">Managing labels - GitHub Docs</a></h3>

  <p>You can classify issues, pull requests, and discussions by creating, editing, applying, and deleting labels.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>say, if you created an Issue template with a label, and non-write access users can simply create it through your template, then, that will bypass the limitation</p>
<p>again, that’s is just what I assume given the limitation, but maybe, they will be seeing this post and consider this as a feature request,</p>

**Comment by sarina** [comment, marked as answer]
<p><a class="mention" href="/jdevstatic">@jdevstatic</a> thanks for the quick reply!</p>
<p>We did some testing last night and figured it indeed must be the limitation of who can add labels. Only people with write access could create the label, even though the template defined it. I think because the label is still added from the user’s permission as an action post-ticket creation (that’s my guess from this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/c/ccf329e2feb9373aedae74ab514c509641b458a0.png" data-download-href="/uploads/short-url/tf4nxhabQqBOYcFc0OwOT5c8uDm.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1985317/181071564-51ab30e3-0a9e-41b5-8de7-a1daa9f2d379.png" alt="image" data-base62-sha1="tf4nxhabQqBOYcFc0OwOT5c8uDm" width="345" height="88" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/c/ccf329e2feb9373aedae74ab514c509641b458a0_2_345x88.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/c/ccf329e2feb9373aedae74ab514c509641b458a0_2_517x132.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/c/ccf329e2feb9373aedae74ab514c509641b458a0_2_690x176.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/c/ccf329e2feb9373aedae74ab514c509641b458a0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">886×226 34.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>where arbrandes is the issue reporter, and also has write access.)</p>
<p>For anyone reading this looking for a solution: I plan to use <a href="https://github.com/actions-ecosystem/action-add-labels" class="inline-onebox" rel="noopener nofollow ugc">GitHub - actions-ecosystem/action-add-labels: 🏷️ GitHub Action to add labels</a> to parse the title and apply a label appropriately.</p>
